apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/App.kt

```diff
@@ -194,24 +193,27 @@ fun AndroidScreen(settingsState: SettingsViewState) {
       StartPartOfScreen(settingsState)
     }
     val scope = rememberCoroutineScope()
-    val onComposed: () -> Unit = {
+    val onComposed: suspend (chatId: String?) -> Unit = { chatId ->
+      // coroutine, scope and join() because:
+      // - it should be run from coroutine to wait until this function finishes
+      // - without using scope.launch it throws CancellationException when changing user
+      // - join allows to wait until completion
       scope.launch {
         offset.animateTo(
-          if (chatModel.chatId.value == null) 0f else maxWidth.value,
+          if (chatId == null) 0f else maxWidth.value,
           chatListAnimationSpec()
         )
-        if (offset.value == 0f) {
-          currentChatId = null
-        }
-      }
+      }.join()
     }
     LaunchedEffect(Unit) {
       launch {
         snapshotFlow { chatModel.chatId.value }
           .distinctUntilChanged()
           .collect {
-            if (it != null) currentChatId = it
-            else onComposed()
+            if (it == null) {
+              onComposed(null)
+            }
+            currentChatId = it
```

Comment:
```suggestion
            if (it == null) onComposed(null)
            currentChatId = it
```

hm, so the problem was that it was not resetting currentChatId to null?

Reply:
No, the problem is much larger that this line. When the app setting chatId to null, and after a moment it sets chatId to some chat.id, the animation was still in progress and the way of setting currentChatId was wrong. So it just didn't react on re-setting chatId if this resetting was fast. Not it listens for every change and reacts accordingly.